### PR TITLE
Throw EntityNotFoundException on findById and findByField

### PR DESCRIPTION
**Pull request by sircoderin**

### CODE REVIEW

File: app/dot/cpp/core/exceptions/EntityNotFoundException.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package dot.cpp.core.exceptions;
+
+public class EntityNotFoundException extends Exception {
+  public EntityNotFoundException() {
+    super("general.recordNotFound");
```
Comment (andrewtookay) [inline app/dot/cpp/core/exceptions/EntityNotFoundException.java:5]: translate its message to Romanian

File: app/dot/cpp/core/services/EntityService.java [hunk 2]
```diff
@@ -22,12 +23,24 @@ public EntityService(BaseRepository<T> repository, Config config) {
...
+    final var entity = repository.findById(id);
+    if (entity == null) {
+      throw new EntityNotFoundException();
+    }
+    return entity;
   }
 
-  public T findByField(String field, String value) {
-    return repository.findByField(field, value);
+  public T findByField(String field, String value) throws EntityNotFoundException {
```
Comment (andrewtookay) [inline app/dot/cpp/core/services/EntityService.java:38]: add check for null or empty field and value as well

File: app/dot/cpp/core/services/EntityService.java [hunk 3]
```diff
@@ -92,7 +105,8 @@ public void delete(T entity) {
     repository.delete(entity);
   }
 
-  public <S extends BaseRequest> S getRequest(String id, S request, BiConsumer<S, T>... consumers) {
+  public <S extends BaseRequest> S getRequest(String id, S request, BiConsumer<S, T>... consumers)
+      throws EntityNotFoundException {
```
Comment (andrewtookay) [inline app/dot/cpp/core/services/EntityService.java:109]: this method first checks the id, the same checks happen again in findById, we could surround findById in a try-catch and on catch return the request

**Review by andrewtookay — CHANGES_REQUESTED** [review]

File: app/dot/cpp/core/exceptions/EntityNotFoundException.java (hunk 1 above)
Comment (sircoderin) [reply app/dot/cpp/core/exceptions/EntityNotFoundException.java:5]: It was done in terra as there we store the messages

**Review by andrewtookay — APPROVED** [review]

**Pull request merged**